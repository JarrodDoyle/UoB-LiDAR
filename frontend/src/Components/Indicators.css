.percentage-indicator{
  display: flex;
  flex-direction: row;
  flex-wrap: none;
  align-items: flex-start;
}

.percentage-indicator h3 {
  margin-right: 15px;
}

.percentage-indicator i {
  border-radius: 100%;
  color: var(--white);
  font-size: 0.75em;
  min-width: 28px;
  min-height: 28px;
  width: 28px;
  height: 28px;
  max-width: 28px;
  max-height: 28px;
  line-height: 28px;
  text-align: center;
  margin-left: auto;
}

.percentage-indicator .ok{
  background-color: var(--green);
}

.percentage-indicator .almost{
  background-color: var(--orange);
}

.percentage-indicator .bad{
  background-color: var(--red);
}
