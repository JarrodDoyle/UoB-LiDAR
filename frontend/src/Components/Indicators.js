import React from "react";
import "./Indicators.css";

export function PercentageIndicator(props){
  let names = ["fas fa-check ok", "fas fa-bacon almost", "fas fa-times bad"]
  let style = names[props.percentage === 100 ? 0 : props.percentage >= 60 ? 1 : 2];
  return (
    <div {...props} className="percentage-indicator">
      <i className={style}></i>
    </div>
  );
}
