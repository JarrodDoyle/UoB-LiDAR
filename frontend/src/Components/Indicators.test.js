import React from 'react';
import { PercentageIndicator } from './Indicators.js';
import renderer from 'react-test-renderer';

test('Percentage Indicator bad lower bound', () => {
  const component = renderer.create(
    <PercentageIndicator percentage={0}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Percentage Indicator bad upper bound', () => {
  const component = renderer.create(
    <PercentageIndicator percentage={59}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Percentage Indicator almost lower bound', () => {
  const component = renderer.create(
    <PercentageIndicator percentage={60}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Percentage Indicator almost upper bound', () => {
  const component = renderer.create(
    <PercentageIndicator percentage={99}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Percentage Indicator okay', () => {
  const component = renderer.create(
    <PercentageIndicator percentage={100}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
