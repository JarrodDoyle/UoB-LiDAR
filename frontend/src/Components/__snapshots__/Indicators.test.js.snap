// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Percentage Indicator almost lower bound 1`] = `
<div
  className="percentage-indicator"
  percentage={60}
>
  <i
    className="fas fa-bacon almost"
  />
</div>
`;

exports[`Percentage Indicator almost upper bound 1`] = `
<div
  className="percentage-indicator"
  percentage={99}
>
  <i
    className="fas fa-bacon almost"
  />
</div>
`;

exports[`Percentage Indicator bad lower bound 1`] = `
<div
  className="percentage-indicator"
  percentage={0}
>
  <i
    className="fas fa-times bad"
  />
</div>
`;

exports[`Percentage Indicator bad upper bound 1`] = `
<div
  className="percentage-indicator"
  percentage={59}
>
  <i
    className="fas fa-times bad"
  />
</div>
`;

exports[`Percentage Indicator okay 1`] = `
<div
  className="percentage-indicator"
  percentage={100}
>
  <i
    className="fas fa-check ok"
  />
</div>
`;
